File: apps/authentication/src/service/UserService.ts
'use strict'

import {UserDao} from "../dao/UserDao";

class UserService {
    private userDao: UserDao;

    constructor() {
        this.userDao = new UserDao()
    }

    async createUser(id: string, phoneNumber: string) {
        try {
            console.log("Creating user", id, phoneNumber)
            return await this.userDao.createUser(id, phoneNumber)
        } catch (e) {
            console.log(`Failed to create user`, e)
            throw e
        }
    }

    async confirmUserEmail(id: string) {
        try {
            console.log("Confirming user email", id)
            return await this.userDao.confirmUserEmail(id)
        } catch (e) {
            console.log(`Failed to confirm user email`, e)
            throw e
        }
    }
}

export {UserService}

File: apps/authentication/src/service/user_service.py
from src.dao.user_dao import UserDao


class UserService:
    def __init__(self):
        self.user_dao = UserDao()

    def create_user(self, id: str, phone_number: str):
        self.user_dao.create_user(id=id, phone_number=phone_number)

    def confirm_user_email(self, id: str):
        self.user_dao.confirm_user_email(id=id)
